pkg/operator/resources/operator/operator.go

```diff
@@ -277,285 +285,71 @@ func createCRD(args *FactoryArgs) []runtime.Object {
-											Enum: []extv1.JSON{
-												{
-													Raw: []byte(`"RemoveWorkloads"`),
-												},
-												{
-													Raw: []byte(`"BlockUninstallIfWorkloadsExist"`),
-												},
-											},
-										},
-									},
-									Type:        "object",
-									Description: "CDISpec defines our specification for the CDI installation",
-								},
-								"status": {
-									Type:        "object",
-									Description: "CDIStatus defines the status of the CDI installation",
-									Properties: map[string]extv1.JSONSchemaProps{
-										"targetVersion": {
-											Description: "The desired version of the CDI resource",
-											Type:        "string",
-										},
-										"observedVersion": {
-											Description: "The observed version of the CDI resource",
-											Type:        "string",
-										},
-										"operatorVersion": {
-											Description: "The version of the CDI resource as defined by the operator",
-											Type:        "string",
-										},
-										"phase": {
-											Description: "CDIPhase is the current phase of the CDI deployment",
-											Type:        "string",
-										},
-										"conditions": {
-											Description: "A list of current conditions of the CDI resource",
-											Type:        "array",
-											Items: &extv1.JSONSchemaPropsOrArray{
-												Schema: &extv1.JSONSchemaProps{
-													Type:        "object",
-													Description: "Condition represents the state of the operator's reconciliation functionality.",
-													Properties: map[string]extv1.JSONSchemaProps{
-														"lastHeartbeatTime": {
-															Type:   "string",
-															Format: "date-time",
-														},
-														"lastTransitionTime": {
-															Type:   "string",
-															Format: "date-time",
-														},
-														"message": {
-															Type: "string",
-														},
-														"reason": {
-															Type: "string",
-														},
-														"status": {
-															Type: "string",
-														},
-														"type": {
-															Description: "ConditionType is the state of the operator's reconciliation functionality.",
-															Type:        "string",
-														},
-													},
-													Required: []string{
-														"status",
-														"type",
-													},
-												},
-											},
-										},
-									},
-								},
-							},
-							Required: []string{
-								"spec",
-							},
-						},
-					},
-					AdditionalPrinterColumns: []extv1.CustomResourceColumnDefinition{
-						{Name: "Age", Type: "date", JSONPath: ".metadata.creationTimestamp"},
-						{Name: "Phase", Type: "string", JSONPath: ".status.phase"},
-					},
-					Subresources: &extv1.CustomResourceSubresources{},
+	// Allow overriding working directory for unit tests which run in their own directory and module
+	cwd := os.Getenv("CDI_DIR")
```

Comment:
Can we put this in its own function that takes a groupKind as parameter? Seems like it could work for other CRDs as well, and if it is a function we can call it from other places.